terraform/modules/gcp/secret-manager/main.tf

```diff
@@ -0,0 +1,26 @@
+data "google_project" "project" {}
+
+resource "google_service_account" "service_account" {
+  account_id   = "e2e-test-user"
+  display_name = "KEDA e2e test user"
+  project = data.google_project.project.project_id
+}
+
+resource "google_service_account_key" "credentials" {
+  service_account_id = google_service_account.service_account.name
+}
+
+provider "google" {
+  credentials = google_service_account_key.my_service_account_key.private_key
+  project     = data.google_project.project.project_id
+}
+
+resource "google_secret_manager_secret" "connection_string" {
+  secret_id = "connectionString"
+}
+
+resource "google_secret_manager_secret_version" "connection_string_version" {
+  secret = google_secret_manager_secret.connection_string.id
+
+  secret_data = "postgresql://test-user:test-password@postgresql.gcp-secret-manager-test-ns.svc.cluster.local:5432/test_db?sslmode=disable"
+}
```

Comment:
This could be outdated with the code because it uses a connection string managed from terraform by used by code. 
In Azure Key Vault we pass the real azure service and the e2e test is based on using that service. We can use the same approach here (the scaler type isn't important at all as we are testing the secret-provider and not the scaler).
- Passing secrets through terraform:
	- https://github.com/kedacore/testing-infrastructure/blob/main/terraform/main.tf#L138-L143
	- https://github.com/kedacore/testing-infrastructure/blob/main/terraform/modules/azure/key-vault/vars.tf#L16-L19
	- https://github.com/kedacore/testing-infrastructure/blob/main/terraform/modules/azure/key-vault/main.tf#L60-L65
- The test:
	- https://github.com/kedacore/keda/blob/main/tests/secret-providers/azure_keyvault/azure_keyvault_test.go

Or maybe even better, if the GCP Secret Manager can be updated using code, we could set the secret on each test execution to be always up to date.

WDYT?

Reply:
We should be able to manipulate the secret from the test itself, given that we have access to the SA key. In that case, the test will also include storing the secret value as part of the E2E prior to its consumption. Is this correct?

Reply:
I think that it'll be the best option as it's more replicable. I mean, with terraform we will enable the API, but if we can do it just as part of the test (I guess that this is something fast, if it requires a long time, terraform is worth) it can be replicated without by anyone with the API enabled just executing the test.
WDYT?

Reply:
You're right, updating the secret is a quick process and it should be feasible to do so as part of the test.

Reply:
so, we just need to enable the api to set the secret during the test :) 
Could you remove the other changes (just keep the apis related change)? I'll merge it to enable the secrets API